cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1372,6 +1374,75 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 	return true, nil
 }
 
+func generateMountsModeRunCloudimgRootfsDraft1(mst *initramfsMountsState) error {
+	// 1. mount ESP
+	// TODO cpc images should maybe set partlabel on the UEFI parition
+	if err := mountPartitionMatchingKernelDisk(boot.InitramfsUbuntuSeedDir, "UEFI"); err != nil {
+		return err
+	}
+
+	// get the disk that we mounted the ESP from as a reference
+	// point for future mounts
+	disk, err := disks.DiskFromMountPoint(boot.InitramfsUbuntuSeedDir, nil)
+	if err != nil {
+		return err
+	}
+
+	// 3.1. measure model
+	err = stampedAction("run-model-measured", func() error {
+		return secbootMeasureSnapModelWhenPossible(mst.UnverifiedBootModel)
```

Comment:
this mst.UnverifiedBootModel helper will read the model from ubuntu-boot which this code is not mount even afaict, also I suppose there won't be separate boot dir here?

Reply:
also that code both in snapd and secboot ATM won't measure the fact this model is classic but very likely should

Reply:
measuring `classic: true` vs `classic: false` would be nice. But i guess change of the measurement which is not backwards compatible? I guess it should measure if classic is true.

Yeah,  I don't know how to specify that all my partition layouts are weird. Cause model in my case should be read from ESP (aka ubuntu-seed in uc20 terms, but UEFI in cloudimg-rootfs terms)

Reply:
we can work something in secboot here and secboot the package about the measuring.

About the partitions and layout, that's more annoying snapd codebase assumes some things are fixed and uses global values for them.

Reply:
I can bind mount ESP as ubuntu-seed and ubuntu-boot if that makes things simpler. Also I somehow thought that model is read from either boot or seed....  How do we measure model in install mode?

Reply:
oh, i see what is happening now.

I'm now wondering if i should measure the model from ESP, or if I should include a model in the classic initrd and measure that.

Reply:
given that we use archive signing key, not uc20 one. There is no concern about not shipping or measuring models for classic right now. Especially since those are not defined yet (aka classic:true grade:secured)